// react-fc/createEventHandler.js
"use strict";

exports.__esModule = true;
exports["default"] = exports.createEventHandlerWithConfig = void 0;

var _changeEmitter = require("change-emitter");

var _setObservableConfig = require("./setObservableConfig");

var createEventHandlerWithConfig = function createEventHandlerWithConfig(config) {
  return function () {
    var emitter = (0, _changeEmitter.createChangeEmitter)();
    var stream = config.fromESObservable({
      subscribe: function subscribe(observer) {
        var unsubscribe = emitter.listen(function (value) {
          return observer.next(value);
        });
        return {
          unsubscribe: unsubscribe
        };
      },
      '@@observable': function observable() {
        return this;
      }
    });
    return {
      handler: emitter.emit,
      stream: stream
    };
  };
};

exports.createEventHandlerWithConfig = createEventHandlerWithConfig;
var createEventHandler = createEventHandlerWithConfig(_setObservableConfig.config);
var _default = createEventHandler;
exports["default"] = _default;
